pkg/prometheus/statefulset.go

```diff
@@ -680,27 +680,32 @@ func makeStatefulSetSpec(p monitoringv1.Prometheus, c *Config, ruleConfigMapName
 			thanosImage = *p.Spec.Thanos.Image
 		}
 
+		thanosVersion := semver.MustParse(strings.TrimPrefix(*p.Spec.Thanos.Version, "v"))
+		clusterSupport := thanosVersion.LT(semver.MustParse("0.5.0-rc.0"))
+
 		thanosArgs := []string{
 			"sidecar",
 			fmt.Sprintf("--prometheus.url=http://%s:9090%s", c.LocalHost, path.Clean(webRoutePrefix)),
 			fmt.Sprintf("--tsdb.path=%s", storageDir),
-			fmt.Sprintf("--cluster.address=[$(POD_IP)]:%d", 10900),
 			fmt.Sprintf("--grpc-address=[$(POD_IP)]:%d", 10901),
 		}
 
-		if p.Spec.Thanos.Peers != nil {
-			thanosArgs = append(thanosArgs, fmt.Sprintf("--cluster.peers=%s", *p.Spec.Thanos.Peers))
-		}
-		if p.Spec.Thanos.ClusterAdvertiseAddress != nil {
-			thanosArgs = append(thanosArgs, fmt.Sprintf("--cluster.advertise-address=%s", *p.Spec.Thanos.ClusterAdvertiseAddress))
-		}
-		if p.Spec.Thanos.GrpcAdvertiseAddress != nil {
-			thanosArgs = append(thanosArgs, fmt.Sprintf("--grpc-advertise-address=%s", *p.Spec.Thanos.GrpcAdvertiseAddress))
+		if clusterSupport {
+			thanosArgs = append(thanosArgs, fmt.Sprintf("--cluster.address=[$(POD_IP)]:%d", 10900))
+			if p.Spec.Thanos.Peers != nil {
+				thanosArgs = append(thanosArgs, fmt.Sprintf("--cluster.peers=%s", *p.Spec.Thanos.Peers))
+			}
+			if p.Spec.Thanos.ClusterAdvertiseAddress != nil {
+				thanosArgs = append(thanosArgs, fmt.Sprintf("--cluster.advertise-address=%s", *p.Spec.Thanos.ClusterAdvertiseAddress))
+			}
+			if p.Spec.Thanos.GrpcAdvertiseAddress != nil {
+				thanosArgs = append(thanosArgs, fmt.Sprintf("--grpc-advertise-address=%s", *p.Spec.Thanos.GrpcAdvertiseAddress))
```

Comment:
we still need this even without clustering

Reply:
The grpc listen address (--grpc-address) is still provided unconditionally in the `thanosArgs` initialization, but advertising addresses are part of the gossip protocol (to tell other members how to contact you when joining).

--grpc-advertise-address removed: https://github.com/improbable-eng/thanos/commit/37f89adfd678c0e263a136da34aafe213e88bc24#diff-78f75f79fe3e933fafa9b97d9c6db643L49
--grpc-address remains: https://github.com/improbable-eng/thanos/commit/37f89adfd678c0e263a136da34aafe213e88bc24#diff-78f75f79fe3e933fafa9b97d9c6db643R19

Reply:
:man_facepalming: you're totally right, sorry